### PR TITLE
platform/1814: Add HSTS header in the definition of `catch-all-server.conf` and `from-to-www.conf` servers.

### PR DESCRIPTION
**Pull request by filippolmt**

### CODE REVIEW

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -20,6 +20,26 @@ else
...
+    export NGINX_HSTS_HEADER="${NGINX_HSTS_HEADER}; includeSubDomains"
+  fi
+  if [ "${NGINX_HSTS_PRELOAD}" -eq 1 ]; then
+    export NGINX_HSTS_HEADER="${NGINX_HSTS_HEADER}; preload"
+  fi
+  sed -e '/#hstsheader/r /templates/hsts.conf' -i /templates/default.conf;
+  sed -e '/#hstsheader/r /templates/hsts.conf' -i /templates/subfolder.conf;
+  sed -e '/#hstsheader/r /templates/hsts.conf' -i /templates/catch-all-server.conf;
+  sed -e '/#hstsheader/r /templates/hsts.conf' -i /templates/from-to-www.conf.tpl;
+fi
```
Comment (paolomainardi) [inline docker-entrypoint.sh:41]: why having the fragment isn't enough?

Comment (paolomainardi) [reply docker-entrypoint.sh:41]: I guess i got it, the problem is that, fragments are actually loaded only by:

1. default.conf
2. subfolder.conf

And not by other server definitions like:

1. catch-all-server.conf
2. from-to-www.conf

That's why you are expanding this approach by using a sed substitution.

I am wondering how we can scale this approach and if we should instead just rely on fragments and assume that all Nginx servers defined by us inherit the fragments as well, even tho some fragments (like osb.conf) are not really needed nor useful in *not-apps* nginx server, like catch-all-server.

Comment (paolomainardi) [reply docker-entrypoint.sh:41]: Anyway the implementation is quite straightforward now, i ask you to:

1. Update the CHANGELOG file - use this space to explain better what is changing here
2. Update the PR title to make it clearer ("Add hosts header on redirects" is not very related)

Comment (filippolmt) [reply docker-entrypoint.sh:41]: Changelog updated and PR title changed

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,17 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## 2022-12-12
+
+### Added
+
+Added HSTS header in the server definition `catch-all-server.conf` and `from-to-www.conf`, if enabled it will automatically add the header in the following servers:
```
Comment (paolomainardi) [inline CHANGELOG.md]: Add a * to make a list

Comment (filippolmt) [reply CHANGELOG.md]: added

**Pull request merged**